packages/pds/src/auth-verifier.ts

```diff
@@ -484,18 +481,38 @@ export class AuthVerifier {
         { audience: [this.dids.pds] },
       )
 
-      const { sub } = result.claims
+      const { sub, scope } = result.claims
       if (typeof sub !== 'string' || !sub.startsWith('did:')) {
         throw new InvalidRequestError('Malformed token', 'InvalidToken')
       }
 
+      if (!scope?.includes('transition:generic')) {
+        throw new AuthRequiredError('Missing required scope', 'InvalidToken')
+      }
+
+      const scopeEquivalent = scope.includes('transition:chat.bsky')
```

Comment:
This means of checking the scopes probably works fine today, but we could run into some issues with it.  E.g. if you can get the AS to hand out a scope it doesn't know about e.g. `unknown:transition:chat.bsky` then this check would be satisfied.  Or if down the road you can further qualify/attenuate legitimate scopes as prefixes/suffixes e.g. `transition:chat.bsky.convo.getLog`.

Reply:
Would support parsing it into an array or set to makes these checks more precise.

Reply:
I usually parse those into an array/Set. Missed this one.